src/RestControllers/AppointmentRestController.php

```diff
@@ -40,10 +75,32 @@ public function getOneForPatient($auuid, $patientUuid)
         return RestControllerHelper::responseHandler($data[0] ?? [], null, 200);
     }
 
-    public function getAll()
+    /**
+     * Retrieves all appointments based on the provided filters.
+     *
+     * @param array $data An array containing the filter criteria.
+     * @return mixed Returns the result of retrieving appointments based on the filters.
+     */
+    public function getAll(array $data)
```

Comment:
if is possible add the return type:
public function getAll(array $data): TypeToReturn